rust/cymbal/src/symbol_store/caching.rs

```diff
@@ -0,0 +1,121 @@
+use std::{collections::HashMap, sync::Arc, time::Instant};
+
+use axum::async_trait;
+use tokio::sync::Mutex;
+
+use crate::{
+    error::Error,
+    metric_consts::{
+        STORE_CACHED_BYTES, STORE_CACHE_EVICTIONS, STORE_CACHE_HITS, STORE_CACHE_MISSES,
+        STORE_CACHE_SIZE,
+    },
+};
+
+use super::{SymbolSetRef, SymbolStore};
+
+pub struct CachingStore {
+    inner: Box<dyn SymbolStore>,
+    cache: Mutex<CacheInner>,
```

Comment:
I assume the cache persists as long as the reference to it is held. Not a concern for now but I'm guessing that means it needs to be repopulated if we redeploy etc... or is the intention to have a further persistent key-value store on top of this in memory cache?

Reply:
Yeah this is all purely in memory - we can add more layers if/when we want (I expect we'll at least want an S3 layer to store sourcemaps in to prevent us relying on the outside world more than we have to).

We'll also be doing per-frame caching in the `Resolver`, and maybe persisting that somewhere, and then of course once we actually resolve a full stack trace we'll store that somewhere for the slow-lane/fast-lane thing